zkstack_cli/crates/zkstack/src/commands/chain/gateway/gateway_common.rs

```diff
@@ -210,17 +212,19 @@ pub(crate) async fn get_gateway_migration_state(
             ));
         }
 
-        let current_sl_from_server = gateway_migration_status.settlement_layer.chain_id().0;
+        let Some(current_sl_from_server) = gateway_migration_status.settlement_layer else {
+            return Ok(GatewayMigrationProgressState::UndefinedSettlementLayer);
```

Comment:
can you please just write `todo!` here, I will do a separate PR that fixes zkstack_cli (other parts of this file are a bit buggy as well, but we definetely should not add `UndefinedSettlementLayer` here, we should give users clarity as much as possible and it is possible to give it here since we have direction of migration